nanoFramework.WebServer/HttpMultipartParser/MultipartFormDataParser.cs

```diff
@@ -0,0 +1,293 @@
+
+                stream.Write(line, 0, line.Length);
+            }
+        }
+
+        private static void TrimEndline(MemoryStream stream)
+        {
+            if (stream.Length == 0)
+            {
+                return;
+            }
+
+            stream.Position = stream.Length - 1;
+
+            int b = stream.ReadByte();
+            int clip = 0;
+
+            if (b == '\n')
+            {
+                clip = 1;
+
+                if (stream.Length > 1)
+                {
+                    stream.Position = stream.Length - 2;
+                    b = stream.ReadByte();
+
+                    if (b == '\r')
+                    {
+                        clip = 2;
+                    }
+                }
+            }
+
+            if (clip > 0)
+            {
+                stream.SetLength(stream.Length - clip);
+            }
+        }
+
+        private void ParseParameterPart(Hashtable parameters, LineReader reader)
+        {
+            StringBuilder sb = new();
+
+            while (true)
+            {
+                byte[] line = reader.ReadByteLine();
+
+                if (line == null || CheckForBoundary(line))
+                {
+                    _parameters.Add(new ParameterPart(parameters["name"].ToString(), sb.ToString()));
+                    break;
+                }
+
+                sb.Append(Encoding.UTF8.GetString(line, 0, line.Length));
+            }
+        }
+
+        private void SkipPart(LineReader reader)
+        {
+            while (true)
+            {
+                byte[] line = reader.ReadByteLine();
+                if (line == null || CheckForBoundary(line))
+                    break;
+            }
+        }
+
+        private bool CheckForBoundary(byte[] line)
+        {
+            if (line == null)
+            {
+                _readEndBoundary = true;
+                return true;
+            }
+
+            int length = _boundaryBinary.Length;
+
+            if (line.Length < length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < length; i++)
+            {
+                if (line[i] != _boundaryBinary[i])
+                {
+                    return false;
+                }
+            }
+
+            //if we get here we have a boundary, check if it is the endboundary
+            if (line.Length >= length + 2 && line[length] == '-' && line[length + 1] == '-')
+            {
+                _readEndBoundary = true;
+            }
+
+            return true;
+        }
+
+        private void RemoveWellKnownParameters(Hashtable parameters)
+        {
+            string[] wellKnownParameters = new[] { "name", "filename", "filename*", "content-type", "content-disposition" };
+
+            foreach (string parameter in wellKnownParameters)
+                if (parameters.Contains(parameter))
+                    parameters.Remove(parameter);
+        }
+    }
+
+    /// <summary>
```

Comment:
please move this MultipartFormDataParserException in a new file, we have 1 file per class/enum/etc

Reply:
Done